editor/plugins/tool_button_editor_plugin.cpp

```diff
@@ -0,0 +1,88 @@
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "tool_button_editor_plugin.h"
+
+#include "editor/editor_node.h"
+#include "editor/editor_property_name_processor.h"
+#include "editor/editor_undo_redo_manager.h"
+#include "scene/gui/button.h"
+
+bool ToolButtonInspectorPlugin::can_handle(Object *p_object) {
+	Ref<Script> scr = Object::cast_to<Script>(p_object->get_script());
+	return scr.is_valid() && scr->is_tool();
+}
+
+void ToolButtonInspectorPlugin::update_action_icon(Button *p_action_button) {
+	p_action_button->set_icon(p_action_button->get_theme_icon(action_icon, SNAME("EditorIcons")));
+}
+
+void ToolButtonInspectorPlugin::call_action(Object *p_object, const StringName &p_method_name) {
+	bool method_is_valid = false;
+	int arg_count = p_object->get_method_argument_count(p_method_name, &method_is_valid);
+
+	if (!method_is_valid) {
+		print_error(vformat("Tool button method is invalid. Could not find method '%s' on %s", p_method_name, p_object->get_class_name()));
+		return;
+	}
+
+	if (arg_count == 0) {
+		p_object->call(p_method_name);
+	} else {
+		p_object->call(p_method_name, EditorUndoRedoManager::get_singleton());
+	}
+}
+
+bool ToolButtonInspectorPlugin::parse_property(Object *p_object, const Variant::Type p_type, const String &p_path, const PropertyHint p_hint, const String &p_hint_text, const BitField<PropertyUsageFlags> p_usage, const bool p_wide) {
+	if (p_type == Variant::CALLABLE) {
+		if (p_usage.has_flag(PROPERTY_USAGE_INTERNAL) && p_usage.has_flag(PROPERTY_USAGE_EDITOR)) {
```

Comment:
You could reverse these conditions to do early return.

Reply:
PROPERTY_USAGE_INTERNAL is still not removed from here.

Reply:
No, only from here. The property should be internal, but it's irrelevant for the inspector.